tests/unit/test_sftp.py

```diff
@@ -0,0 +1,224 @@
+
+    connector = SftpConnector(credentials=variables["credentials"])
+    connector.get_connection()
+
+    mock_rsa_key.assert_called_once()
+    mock_rsa_key.assert_called_once()
+    mock_ssh_connect.assert_called_once_with(
+        "", username="test_user", pkey=mock_rsa_key.return_value
+    )
+    assert connector.conn is not None
+
+
+@pytest.mark.functions()
+def test_to_df_with_csv(mocker):
+    """Test SFTP `to_df` method with csv."""
+    mock_getfo_file = mocker.patch.object(SftpConnector, "_getfo_file", autospec=True)
+    mock_getfo_file.return_value = BytesIO(b"col1,col2\n1,2\n3,4\n")
+
+    connector = SftpConnector(credentials=variables["credentials"])
+    df = connector.to_df(file_name="test.csv", sep=",")
+
+    assert isinstance(df, pd.DataFrame)
+    assert df.shape == (2, 4)
+    assert list(df.columns) == [
+        "col1",
+        "col2",
+        "_viadot_source",
+        "_viadot_downloaded_at_utc",
+    ]
+
+
+@pytest.mark.functions()
+def test_to_df_with_tsv(mocker):
+    """Test SFTP `to_df` method with tsv."""
+    mock_getfo_file = mocker.patch.object(SftpConnector, "_getfo_file", autospec=True)
+    mock_getfo_file.return_value = BytesIO(b"col1,col2\n1,2\n3,4\n")
+
+    connector = SftpConnector(credentials=variables["credentials"])
+    df = connector.to_df(file_name="test.tsv", sep=",")
+
+    assert isinstance(df, pd.DataFrame)
+    assert df.shape == (2, 4)
+    assert list(df.columns) == [
+        "col1",
+        "col2",
+        "_viadot_source",
+        "_viadot_downloaded_at_utc",
+    ]
+
+
+@pytest.mark.functions()
+def test_to_df_with_json(mocker):
+    """Test SFTP `to_df` method with json."""
+    mock_getfo_file = mocker.patch.object(SftpConnector, "_getfo_file", autospec=True)
+    json_data = json.dumps({"col1": [1, 3], "col2": [2, 4]})
+    mock_getfo_file.return_value = BytesIO(json_data.encode("utf-8"))
+
+    connector = SftpConnector(credentials=variables["credentials"])
+    df = connector.to_df(file_name="test.json")
+
+    expected_df = pd.DataFrame({"col1": [1, 3], "col2": [2, 4]})
+    expected_df["_viadot_source"] = "SftpConnector"
+    expected_df["_viadot_downloaded_at_utc"] = pd.Timestamp.now()
+
+    assert isinstance(df, pd.DataFrame)
+    assert list(df.columns) == [
+        "col1",
+        "col2",
+        "_viadot_source",
+        "_viadot_downloaded_at_utc",
+    ]
+
+
+@pytest.mark.functions()
+def test_list_directory(mocker):
+    """Test SFTP `_list_directory` method."""
+    mock_sftp = mocker.MagicMock()
+    mock_sftp.listdir.return_value = ["file1.txt", "file2.txt"]
+
+    connector = SftpConnector(credentials=variables["credentials"])
+    connector.conn = mock_sftp
+    files_list = connector._list_directory()
+
+    assert files_list == ["file1.txt", "file2.txt"]
+    mock_sftp.listdir.assert_called_once_with(".")
+
+
+@pytest.mark.functions()
+def test_recursive_listdir(mocker):
+    """Test SFTP `_recursive_listdir` method."""
```

Comment:
Not sure how this is testing the recursiveness? For that you need a dir and a subdir, and you only have the top-level dir, with no subdir? Can you briefly explain the logic of this test in the docstring, because it's not easy to follow?

Reply:
I did because the test coverage need to reach over 80%. It is passing the test, so if you think we need to change something, please add it to a backlog task, explaining the problem for a future review.

Reply:
Updated test